.github/workflows/publish-helm-chart.yml

```diff
@@ -74,6 +80,66 @@ jobs:
           - name: Helm Chart Linter
             run: |
               helm lint ${{matrix.chartpath}}
+            
+          - name: SEMVER Check
+            #Checks Chart Version Number against SEMVER regex
+            run: | 
+              version=$(helm show chart ${{matrix.chartpath}} | 
+              awk '/version:/ {print $2}')
+
+              semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\
+              (-(0|[1-9A-Za-z-][0-9A-Za-z-]*)(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]\
+              +(\.[0-9A-Za-z-]+)*)?$"
+
+              if [[ $version =~ $semver_regex ]]; then
+                echo "$version uses SEMVER, https://semver.org/"
+              else
+                echo "$version is not using SEMVER, https://semver.org/"
+                exit 1
+              fi 
+
+          - name: Helm Chart Version Check
+            env: 
+              BASE: ${{needs.build-chart-list.outputs.BASE}}
+            run: |
+              # old_chart_version diffs commit that triggered workflow against previous 
+              # commit. Pulls out version number of the existing chart or "old chart".
+              # The git diff gives spits out both version numbers, head -2 | tail -1
+              # Grabs the value on the second line. 
+              old_chart_version=$(git diff "$BASE" \
+               ${{ github.sha }} -- ${{matrix.chartpath}} | 
+               awk '/version:/ {print $2}' | 
+               head -2 | 
+               tail -1 | 
+               tr -d '"')
+
+              # Grabs the chart version number out of the chart that needs
+              # to be checked. 
+              new_chart_version=$(helm show chart ${{matrix.chartpath}} | 
+              awk '/version:/ {print $2}')
+
+              # Takes the version numbers and replaces dots with zeroes.
+              old_version=$(sed 's/\./0/g' <<< "$old_chart_version")
+              new_version=$(sed 's/\./0/g' <<< "$new_chart_version")
```

Comment:
I'm not sure that this always works. What if the old chart version is `1.2.3` and the author writes the new chart version as `1.2.01`?